client/signup/steps/site-type/form.jsx

```diff
@@ -40,52 +36,34 @@ class SiteTypeForm extends Component {
 		};
 	}
 
-	handleRadioChange = event => this.setState( { siteType: event.currentTarget.value } );
-
-	handleSubmit = event => {
-		const { siteType } = this.state;
-
-		event.preventDefault();
-
+	handleSubmit = type => {
 		this.props.recordTracksEvent( 'calypso_signup_actions_submit_site_type', {
-			value: siteType,
+			value: type,
 		} );
 
-		this.props.submitForm( siteType );
+		this.setState( { siteType: type } );
+
+		this.props.submitForm( type );
 	};
 
 	renderRadioOptions() {
 		return allSiteTypes.map( siteTypeProperties => (
-			<FormLabel
-				className={ classNames( 'site-type__option', {
-					'is-selected': siteTypeProperties.slug === this.state.siteType,
-				} ) }
+			<button
```

Comment:
I don't think we need to rediscover the wheel for this: we already have a linkable card component - see https://wpcalypso.wordpress.com/devdocs/design/card - the "I am a linkable card" and "I am a clickable button that looks like a link" examples.